tests/hardhat/test/Token.ts

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable no-undef */
+// We import Chai to use its asserting functions here.
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { expect } from 'chai';
+import { Contract, ContractFactory } from 'ethers';
+import { ethers } from 'hardhat';
+import { transpile, sh, warp_deploy, warp_call } from '../util';
+
+// `describe` is a Mocha function that allows you to organize your tests. It's
+// not actually needed, but having your tests organized makes debugging them
+// easier. All Mocha functions are available in the global scope.
+
+// `describe` receives the name of a section of your test suite, and a callback.
+// The callback must define the tests of that section. This callback can't be
+// an async function.
+describe('Token contract', function () {
+  this.timeout(600_000); // 10 min
+  // Mocha has four functions that let you hook into the the test runner's
+  // lifecyle. These are: `before`, `beforeEach`, `after`, `afterEach`.
+
+  // They're very useful to setup the environment for tests, and to clean it
+  // up after they run.
+
+  // A common pattern is to declare some variables, and assign them in the
+  // `before` and `beforeEach` callbacks.
+
+  let Token: ContractFactory;
+  let hardhatToken: Contract;
+  let owner: SignerWithAddress;
+  // eslint-disable-next-line no-unused-vars
+  let addrs: SignerWithAddress[];
+
+  // Prepend all Starknet/Cairo variables with c_
+  let c_TokenAddress: string;
+
+  beforeEach(function () {});
+
+  // You can nest describe calls to create subsections.
+  describe('Transpilation & Deployment', async function () {
+    it('Transpile Solidity to Cairo', async function () {
+      await transpile('contracts/Token.sol', 'Token');
+      await sh('mv contracts/Token.json contracts/warp_artifacts');
+    });
+
+    it('Deploy Solidity & Cairo Contracts', async function () {
+      // ========================================================
+      // Solidity Deployment
+      // ========================================================
+      const c_pTokenAddress = warp_deploy('Token');
+      Token = await ethers.getContractFactory('Token');
+      [owner, ...addrs] = await ethers.getSigners();
+      const pHardhatToken = Token.deploy();
+
+      // ========================================================
+      // StarkNet Deployment
+      // ========================================================
+      [hardhatToken, c_TokenAddress] = await Promise.all([
+        pHardhatToken,
+        c_pTokenAddress,
+      ]);
+    });
+
+    // Get the ContractFactory and Signers here.
+    // `it` is another Mocha function. This is the one you use to define your
+    // tests. It receives the test name, and a callback function.
```

Comment:
The description of `it` is a bit too late. Perhaps, we should move it before the first `it` function or remove altogether and just give a link to the docs